python/gtsam/tests/test_custom_factor.py

```diff
@@ -0,0 +1,86 @@
+"""
+GTSAM Copyright 2010-2019, Georgia Tech Research Corporation,
+Atlanta, Georgia 30332-0415
+All Rights Reserved
+
+See LICENSE for the license information
+
+CustomFactor unit tests.
+Author: Fan Jiang
+"""
+from typing import List
+import unittest
+from gtsam import Values, Pose2, CustomFactor
+
+import numpy as np
+
+import gtsam
+from gtsam.utils.test_case import GtsamTestCase
+
+
+class TestCustomFactor(GtsamTestCase):
+
+    def test_new(self):
+        def error_func(this: CustomFactor, v: gtsam.Values, H: List[np.ndarray]):
+            return np.array([1, 0, 0])
+        
+        noise_model = gtsam.noiseModel.Unit.Create(3)
+        cf = CustomFactor(noise_model, gtsam.KeyVector([0]), error_func)
```

Comment:
is `gtsam.KeyVector(...)` necessary?  I thought for KeyVector then

```python
cf = CustomFactor(noise_model, [0], error_func)
```

should work?  If not, is there a different type for which this would work?

Reply:
We should consider making KeyVector default to a regular list on the python side while keeping the allocator semantics on the cpp side

Reply:
Now it should work, I added the implicit conversion for `list`.